arches/app/search/es_mapping_modifier.py

```diff
@@ -0,0 +1,80 @@
+from arches.app.models.system_settings import settings
+from arches.app.utils import import_class_from_string
+
+
+class EsMappingModifierFactory:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def get_es_mapping_modifier_classes():
+        return (
+            [
+                import_class_from_string(classname)
+                for classname in settings.ES_MAPPING_MODIFIER_CLASSES
+            ]
+            if settings.setting_exists("ES_MAPPING_MODIFIER_CLASSES")
+            and settings.ES_MAPPING_MODIFIER_CLASSES
+            else []
+        )
+
+
+class EsMappingModifier:
+    """
+    Base class for creating custom sections in the Resource Instance elasticsearch document.
+    """
+
+    custom_search_path = "custom_values"
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def get_custom_search_path():
```

Comment:
```suggestion
    def get_mapping_property():
```

see this [doc example](https://www.elastic.co/guide/en/elasticsearch/reference/current/explicit-mapping.html) of the es mapping property